packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayer.m

```diff
@@ -421,6 +426,216 @@ - (void)setPlaybackSpeed:(double)speed error:(FlutterError *_Nullable *_Nonnull)
   [self updatePlayingState];
 }
 
+- (nullable FVPNativeAudioTrackData *)getAudioTracks:(FlutterError *_Nullable *_Nonnull)error {
+  AVPlayerItem *currentItem = _player.currentItem;
+  if (!currentItem || !currentItem.asset) {
+    return [FVPNativeAudioTrackData makeWithAssetTracks:nil mediaSelectionTracks:nil];
+  }
+
+  AVAsset *asset = currentItem.asset;
+
+  // First, try to get tracks from media selection (for HLS streams)
+  AVMediaSelectionGroup *audioGroup =
+      [asset mediaSelectionGroupForMediaCharacteristic:AVMediaCharacteristicAudible];
+  if (audioGroup && audioGroup.options.count > 0) {
+    // Cache the options array for later use in selectAudioTrack
+    _cachedAudioSelectionOptions = audioGroup.options;
+
+    NSMutableArray<FVPMediaSelectionAudioTrackData *> *mediaSelectionTracks =
+        [[NSMutableArray alloc] init];
+    AVMediaSelectionOption *currentSelection = nil;
+    if (@available(iOS 11.0, macOS 10.13, *)) {
+      AVMediaSelection *mediaSelection = currentItem.currentMediaSelection;
+      currentSelection = [mediaSelection selectedMediaOptionInMediaSelectionGroup:audioGroup];
+    } else {
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+      currentSelection = [currentItem selectedMediaOptionInMediaSelectionGroup:audioGroup];
+#pragma clang diagnostic pop
+    }
+
+    for (NSInteger i = 0; i < audioGroup.options.count; i++) {
+      AVMediaSelectionOption *option = audioGroup.options[i];
+
+      // Skip nil options
+      if (!option || [option isKindOfClass:[NSNull class]]) {
+        continue;
+      }
+
+      NSString *displayName = option.displayName;
+
+      NSString *languageCode = nil;
+      if (option.locale) {
+        languageCode = option.locale.languageCode;
+      }
+
+      NSString *commonMetadataTitle = nil;
+      NSArray<AVMetadataItem *> *titleItems =
+          [AVMetadataItem metadataItemsFromArray:option.commonMetadata
+                                         withKey:AVMetadataCommonKeyTitle
+                                        keySpace:AVMetadataKeySpaceCommon];
+      if (titleItems.count > 0 && titleItems.firstObject.stringValue) {
```

Comment:
nit: is if isn't needed? `firstObject` returns `nil` if the array is empty according to documentation.

Reply:
updated